pinot-spi/src/main/java/org/apache/pinot/spi/queryeventlistener/BrokerQueryEventListener.java

```diff
@@ -0,0 +1,24 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.queryeventlistener;
+
+public interface BrokerQueryEventListener {
```

Comment:
I feel we can put it along with `RequestContext` and call it `QueryEventListener`

Reply:
We are planning to have 2 event listeners - `BrokerQueryEventListener` and `ServerQueryEventListener` so better to keep it separate imo.

Reply:
yeah i felt the same in spi module we should just call QueryEventListener. and for the concrete impl we can call it broker/server prefix

Reply:
But the issue is we don't have any RequestContext like-class for Server. And I don't think there is any server-related stats object in spi package as well. 
I think we might have to go with some server specific object like `ServerQueryEventInfo` for the Server Listener class. So want to keep these broker / server listener separate.

Reply:
Yes exactly!